packages/trace-viewer/src/ui/errorsTab.tsx

```diff
@@ -114,24 +124,66 @@ export const ErrorsTab: React.FunctionComponent<{
         location = file + ':' + stackFrame.line;
         longLocation = stackFrame.file + ':' + stackFrame.line;
       }
-      return <div key={message}>
+
+      const errorId = `error-${wallTime}-${longLocation}`;
+
+      return <div key={message} style={{ minHeight: errorsModel.errors.size === 1 ? '100%' : undefined, display: 'flex', flexDirection: 'column' }}>
         <div className='hbox' style={{
           alignItems: 'center',
           padding: '5px 10px',
           minHeight: 36,
           fontWeight: 'bold',
           color: 'var(--vscode-errorForeground)',
+          flex: 0,
         }}>
           {error.action && renderAction(error.action, { sdkLanguage })}
           {location && <div className='action-location'>
             @ <span title={longLocation} onClick={() => revealInSource(error)}>{location}</span>
           </div>}
           <span style={{ position: 'absolute', right: '5px' }}>
-            <PromptButton error={message} actions={actions} />
+            {llmAvailable
+              ? <ToolbarButton onClick={() => setShowLLM(v => !v)} title="Fix with AI" className='copy-to-clipboard-text-button'>Fix with AI</ToolbarButton>
+              : <CopyPromptButton error={message} pageSnapshot={pageSnapshot} diff={diff} />}
           </span>
         </div>
+        
         <ErrorMessage error={message} />
+
+        {showLLM && <AIErrorConversation error={message} pageSnapshot={pageSnapshot} conversationId={errorId} diff={diff} />}
       </div>;
     })}
   </div>;
 };
+
+export function AIErrorConversation({ conversationId, error, pageSnapshot, diff }: { conversationId: string, error: string, pageSnapshot?: string, diff?: string }) {
+  const [history, conversation] = useLLMConversation(
```

Comment:
We'll have to clean them up at some point. Let's think about it, and if the solution is clear - add a TODO and follow up later.

Perhaps we should store them somewhere in the test run model instead of coming up with ids?

Reply:
You currently can't recover a conversation after re-running tests though.